<a href="https://colab.research.google.com/github/riza93n-hub/data-science-2026/blob/main/Pertemuan2_RIZA_250401020014.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Aktivitas Hands On Pertemuan 2
**Nama Mahasiswa:** RIZA  
**NIM:** 250401020014  
**Kelas:** IF401

In [1]:
import pandas as pd
import numpy as np

# 1. Muat Dataset Titanic dari URL GitHub
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)

# 2. Eksplorasi Awal Data
print("--- 5 Baris Pertama Dataset ---")
display(df.head())
print("\n--- Ukuran Dataset (Baris, Kolom) ---")
display(df.shape)

--- 5 Baris Pertama Dataset ---


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S



--- Ukuran Dataset (Baris, Kolom) ---


(891, 12)

In [2]:
# 1. Daftar nama kolom dalam tabel
print("Daftar nama kolom:")
for col in df.columns:
    print(f"- {col}")

print("\n" + "="*40 + "\n")

# 2. Mengecek kolom yang memiliki data kosong (Missing Values)
missing = df.isna().sum()
missing = missing[missing > 0]

if not missing.empty:
    max_len = max(len(col) for col in missing.index)
    print("Kolom dengan missing values:")
    for col, val in missing.items():
        print(f"- {col.ljust(max_len)} : {val}")
else:
    print("Tidak ada kolom dengan missing values.")

Daftar nama kolom:
- PassengerId
- Survived
- Pclass
- Name
- Sex
- Age
- SibSp
- Parch
- Ticket
- Fare
- Cabin
- Embarked


Kolom dengan missing values:
- Age      : 177
- Cabin    : 687
- Embarked : 2


In [3]:
# 1. Menyaring data penumpang wanita kelas 1 yang selamat
female_class1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)]
survived = female_class1[female_class1['Survived'] == 1]
survived_count = survived['Survived'].count()
avg_age = survived['Age'].mean()

print("Analisis Penumpang Wanita Kelas 1 yang Selamat:")
print(f"- Jumlah yang selamat : {survived_count} orang")
print(f"- Rata-rata usia      : {avg_age:.2f} tahun")

print("\n" + "="*40 + "\n")

# 2. Mencari kelas penumpang dengan tingkat keselamatan tertinggi
survival_rate = df.groupby('Pclass')['Survived'].mean()

print("Tingkat keselamatan per kelas:")
for pclass, rate in survival_rate.items():
    print(f"- Kelas {pclass} : {rate:.2%}")

best_class = survival_rate.idxmax()
best_rate = survival_rate.max()
print(f"\nKelas dengan tingkat keselamatan tertinggi adalah Kelas {best_class} ({best_rate:.2%})")

Analisis Penumpang Wanita Kelas 1 yang Selamat:
- Jumlah yang selamat : 91 orang
- Rata-rata usia      : 34.94 tahun


Tingkat keselamatan per kelas:
- Kelas 1 : 62.96%
- Kelas 2 : 47.28%
- Kelas 3 : 24.24%

Kelas dengan tingkat keselamatan tertinggi adalah Kelas 1 (62.96%)


In [4]:
print("=== HASIL JAWABAN PERTANYAAN EVALUASI ===")

# Q1: Berapa jumlah baris dan kolom dataset Titanic?
rows, cols = df.shape
print(f"Q1. Ukuran dataset Titanic -> Jumlah baris: {rows}, Jumlah kolom: {cols}")

# Q2: Kolom mana yang memiliki nilai kosong (missing)?
missing_cols = df.columns[df.isna().any()]
print(f"Q2. Kolom dengan missing values: {', '.join(missing_cols)}")

# Q3: Berapa persen penumpang yang selamat (Survived=1)?
overall_survival_rate = df['Survived'].mean() * 100
print(f"Q3. Persentase penumpang yang selamat: {overall_survival_rate:.2f}%")

# Q4: Rata-rata usia penumpang wanita kelas 1 yang selamat?
avg_age_female_c1 = df[
    (df['Sex'] == 'female') &
    (df['Pclass'] == 1) &
    (df['Survived'] == 1)
]['Age'].mean()
print(f"Q4. Rata-rata usia penumpang wanita kelas 1 yang selamat: {avg_age_female_c1:.2f} tahun")

# Q5 & Q6: Kelas penumpang mana yang tingkat keselamatan tertinggi beserta rincian kelompoknya?
survival_rate_group = df.groupby('Pclass')['Survived'].mean()
best_class_group = survival_rate_group.idxmax()

print(f"Q5. Kelas dengan tingkat keselamatan tertinggi: Kelas {best_class_group}")
print("Q6. Rincian hasil persentase keselamatan kelompok per kelas:")
for pclass, rate in survival_rate_group.items():
    print(f"    - Kelas {pclass}: {rate:.2%}")

=== HASIL JAWABAN PERTANYAAN EVALUASI ===
Q1. Ukuran dataset Titanic -> Jumlah baris: 891, Jumlah kolom: 12
Q2. Kolom dengan missing values: Age, Cabin, Embarked
Q3. Persentase penumpang yang selamat: 38.38%
Q4. Rata-rata usia penumpang wanita kelas 1 yang selamat: 34.94 tahun
Q5. Kelas dengan tingkat keselamatan tertinggi: Kelas 1
Q6. Rincian hasil persentase keselamatan kelompok per kelas:
    - Kelas 1: 62.96%
    - Kelas 2: 47.28%
    - Kelas 3: 24.24%


## Kesimpulan

* **Apa yang Dipelajari:** Di pertemuan kedua ini, saya belajar cara menggunakan library Pandas di Python untuk membuka dan mengolah file data dari internet. Saya mempraktikkan cara melihat sekilas isi tabel data, mengecek ukuran baris dan kolom, mencari tahu kolom mana saja yang datanya masih kosong, serta menyaring data penumpang berdasarkan kriteria tertentu dan mengelompokkannya untuk menghitung nilai rata-rata.
* **Temuan Utama:** Menggunakan Pandas ternyata membuat proses analisis tabel data yang banyak menjadi jauh lebih cepat dan praktis. Dari dataset Titanic ini, saya menemukan bahwa tidak semua penumpang memiliki data yang lengkap, seperti kolom usia dan nomor kabin yang banyak kosong. Selain itu, dari hasil penyaringan data, terlihat jelas bahwa penumpang kelas 1 memiliki tingkat keselamatan yang jauh lebih tinggi (62.96%) dibandingkan dengan penumpang di kelas 2 dan kelas 3.
* **Keterbatasan / Pertanyaan yang Muncul:** * *Keterbatasan:* Analisis data yang dilakukan di atas baru sebatas menampilkan angka dan teks ringkasan saja, sehingga kita harus membaca deretan angka tersebut dengan teliti untuk melihat perbandingannya.
  * *Pertanyaan:* Bagaimana caranya agar data tabel ini bisa diubah menjadi bentuk grafik atau diagram visual, seperti diagram batang atau lingkaran, supaya hasil perbandingan tingkat keselamatan antar kelasnya bisa langsung dipahami dalam sekali lihat?